glean_parser/templates/ruby_server.jinja2

```diff
@@ -0,0 +1,138 @@
+    else
+      RbConfig::CONFIG['host_os']
+    end
+  end
+  def self.version
+    version = RbConfig::CONFIG['build_os'].match('\d{1,2}\.*\d{0,2}\.*\d{0,3}').to_s
+    if !version.nil?
+        version.to_s
+    else
+        'Unknown'
+    end
+  end
+end
+
+module GleanHelper
+  {# Create a list of ping classes. #}
+  {% for event_category, event_types in events.items() %}
+  {% for event_type, event_object in event_types.items() %}
+  class {{ (event_category + '.' + event_type) | event_class_name }}
+    # see comments for init parameter descriptions
+    def initialize(application_id:, app_display_version:, app_channel:, user_agent:, ip_address:, event_extra)
+      @application_id = application_id # string - Application Id to identify application per Glean standards
+      @app_display_version = app_display_version # string - Application Id to identify application per Glean standards
+      @app_channel = app_channel # string - Application Id to identify application per Glean standards
+      @user_agent = user_agent # string - The user agent per Glean standards
+      @ip_address = ip_address # string - The IP address that will be used to decode Geo information and scrubbed at ingestion per Glean standards
+      @event_extra = event_extra # string - JSON blob containing the backend event payload
+    end
+
+    def record
+      t = Time.now
+      t_utc = t.utc
+      event_payload = {
+        # `Unknown` fields below are required in the Glean schema, however they are not useful in server context.
+        'client_info' => {
+          'telemetry_sdk_build' => 'glean_parser v{{ parser_version }}',
+          'first_run_date' => 'Unknown',
+          'os' => OS.name,
+          'os_version' => OS.version,
+          'architecture' => 'Unknown',
+          'app_build' => 'Unknown',
+          'app_display_version' => @app_display_version,
+          'app_channel' => @app_channel,
+        },
+        'ping_info' => {
+          'seq' => 0, # this is required, however doesn't seem to be useful in server context.
+          'start_time' => t_utc,
+          'end_time' => t_utc,
+        },
+        'event' => '{{ event_category + '.' + event_type }}'
+        'event_timestamp' => t_utc
+        'event_extra' => event_extra,
+      }
+      serialized_event_payload = event_payload.to_json
+      # This is the message structure that Decoder expects: https://github.com/mozilla/gcp-ingestion/pull/2400.
+      ping = {
+        'document_namespace' => @application_id,
+        'document_type' => 'events',
+        'document_version' => '1',
+        'document_id' => SecureRandom.uuid,
+        'user_agent' => @user_agent,
+        'ip_address' => @ip_address,
+        'payload' => serialized_event_payload,
+      }
+      # This structure is similar to how FxA currently logs with mozlog: https://github.com/mozilla/fxa/blob/4c5c702a7fcbf6f8c6b1f175e9172cdd21471eac/packages/fxa-auth-server/lib/log.js#L289
+      logger = Logger.new($stdout)
```

Comment:
Is logging to `$stdout` really what we want? How are applications supposed to configure this?

Reply:
I do not fully understand this question, but I think what you are saying is that we should be sending these logs through whatever logger Mastodon Rails App is using for other types of logs versus hard-coding to stdout.  I can look into this.

Reply:
> (...) we should be sending these logs through whatever logger Mastodon Rails App is using for other types of logs (...)

Yes, best to log to the same destination that application does, but to be fair stdout is probably fine.